File: day-4/day-4.py
import hashlib


def mine_adventcoin(secret_key, num_of_zeros=5):
    number = 1
    target = '0' * num_of_zeros

    while True:
        data = secret_key + str(number)
        result_hash = hashlib.md5(data.encode()).hexdigest()

        if result_hash.startswith(target):
            return number

        number += 1


secretKey = "bgvyzdsv"
print(mine_adventcoin(secretKey))
print(mine_adventcoin(secretKey, 6))

File: day-4/day-4.js
const crypto = require("crypto");

const mineAdventcoin = function (secretKey, numOfZeros = 5) {
  let number = 1;
  const target = "0".repeat(numOfZeros);

  while (true) {
    const data = secretKey + number.toString();
    const hash = crypto.createHash("md5").update(data).digest("hex");

    if (hash.startsWith(target)) {
      return number;
    }
    number++;
  }
};

const secretKey = "bgvyzdsv";

console.log(mineAdventcoin(secretKey, 6));
